pkg/prometheus/operator.go

```diff
@@ -299,3 +303,30 @@ func UpdateServiceMonitorStatus(
 	}
 	return nil
 }
+
+func configResStatusConditionsEqual(a, b []monitoringv1.ConfigResourceCondition) bool {
+	if len(a) != len(b) {
+		return false
+	}
+
+	ac := append([]monitoringv1.ConfigResourceCondition(nil), a...)
+	bc := append([]monitoringv1.ConfigResourceCondition(nil), b...)
+
+	sort.Slice(ac, func(i, j int) bool {
```

Comment:
We need to sort by Type because the API validation ensures that duplicate Type values aren't possible.